Dockerfile

```diff
@@ -1,9 +1,30 @@
-FROM golang:alpine
+# BUILD STEP
+
+FROM golang:alpine as build
+
+RUN set -ex && apk add --no-cache gcc build-base linux-headers
+
+RUN mkdir /build
+WORKDIR /build
+
+COPY go.mod /build
+COPY go.sum /build
+RUN CGO_ENABLED=1 GOOS=linux go mod download
+
+COPY . .
+
+RUN CGO_ENABLED=1 GOOS=linux go build -a -installsuffix cgo -ldflags '-s -w -extldflags "-static"' -o owncast .
+
+# COMMAND STEP
+
+FROM alpine:latest as final
 EXPOSE 8080 1935
+
+RUN set -ex && apk add --no-cache ffmpeg ffmpeg-libs
+	
+RUN mkdir /app
 WORKDIR /app
-ADD . .
-RUN set -ex && \
-    apk add --no-cache ffmpeg ffmpeg-libs && \
-    apk add --no-cache gcc build-base linux-headers && \ 
-    CGO_ENABLED=1 GOOS=linux go build -a -installsuffix cgo -ldflags '-extldflags "-static"' -o owncast .
+COPY --from=build /build/owncast ./owncast
+COPY . .
```

Comment:
In here we could just list all files/directories that need to exist in the final image